### ISSUE TITLE
build-storybook should build stories with "Static Site Generation" (SSG) for improved perf/SEO

### ISSUE THREAD

**Issue by NateRadebaugh** [storybookjs/storybook#9382]
**Is your feature request related to a problem? Please describe.**
I'm always frustrated when I navigate to a page and see the page skeleton while the stories are fetched instead of just showing the full menu/first page immediately. There's already a build step, so this seems possible.

**Describe the solution you'd like**
I'd like to see a solution where static pages are generated from stories without showing the generic storybook skeleton:
![image](https://user-images.githubusercontent.com/130445/72168559-037db300-3393-11ea-9665-57f04dd20721.png)

**Describe alternatives you've considered**
With tools like Gatsby and NextJS bringing support for SSG, I could set up a separate site that utilizes the new CSF format and maybe host it there, but that wouldn't be ideal.

**Are you able to assist bring the feature to reality?**
Yes with guidance since I don't have familiarity with the storybook internals.

**Additional context**
N/A

**Comment by shilman** [comment]
Any interest in taking this on @NateRadebaugh? I think @ndelangen knows the bundling issues better than anyone else...

**Comment by NateRadebaugh** [comment]
Yes I would love to take this on, but I'm not sure where to start.  Any hints?

**Comment by ndelangen** [comment]
@NateRadebaugh Shall we schedule a meeting to talk about this?

You can view my calendar right here:
https://calendly.com/ndelangen/storybook

(and even schedule a meeting there)

Looking forward to hear what you have in mind! I can definitely help you understand what's currently going on in the code and assist with potential changes.

**Comment by kalinchernev** [comment]
Still relevant, not stale

**Comment by muratcorlu** [comment]
I think this becomes a bigger issue if we consider using Storybook as the source of truth in our design system. Currently showing a page [can take 10 seconds](https://component-controls.com/blogs/gatsby-vs-nextjs-vs-storybook). And when we share links in internet, link previews are broken, since we can not set a proper HTML title and meta tags for every separate pages. For example, currently we see something like that in iMessage when we share a storybook link:

<img width="163" alt="image" src="https://user-images.githubusercontent.com/127687/177715265-e0f6b50e-a13f-4b1c-ae3f-b4805d0d7dee.png">[^1]

I can see, that would be a big change for current codebase but I think it should definitely be considered for the next big update.

[^1]: This title will be "Storybook loading..." starting from v7 #17709

**Comment by duannx** [comment]
Hi, is there any update on this feature? Besides SEO, we now have another need: using Storybook as a document feed for AI tools.
